### PR TITLE
Display GenAI analysis in the case page in the Web UI

### PR DESCRIPTION
**Pull request by mvilanova**
This pull request includes several changes to the `Page.vue` and `store.js` files in the `src/dispatch/static/dispatch/src/case` directory. These changes primarily focus on adding support for displaying GenAI analysis, reordering properties for better readability, and importing necessary modules.

### Enhancements to GenAI Analysis Support:

* Added `GenaiAnalysisDisplay` component to the `Page.vue` template for displaying GenAI analysis when the active tab is not 'signals' or 'graph'. (`src/dispatch/static/dispatch/src/case/Page.vue`)
* Included `genai_analysis` property in the `caseDefaults` object and updated the `PageHeader` component to accept this new property. (`src/dispatch/static/dispatch/src/case/Page.vue`) [[1]](diffhunk://#diff-4c33bb2f483c237c20cc9af8f434316809a6e30246c33f1ee5a155a67b369c21L4-R11) [[2]](diffhunk://#diff-4c33bb2f483c237c20cc9af8f434316809a6e30246c33f1ee5a155a67b369c21L46-L94)

### Code Reorganization:

* Reordered properties in the `caseDefaults` object and the `getDefaultSelectedState` function for better readability and consistency. (`src/dispatch/static/dispatch/src/case/Page.vue`, `src/dispatch/static/dispatch/src/case/store.js`) [[1]](diffhunk://#diff-4c33bb2f483c237c20cc9af8f434316809a6e30246c33f1ee5a155a67b369c21L46-L94) [[2]](diffhunk://#diff-4c33bb2f483c237c20cc9af8f434316809a6e30246c33f1ee5a155a67b369c21L176-R183) [[3]](diffhunk://#diff-2713750720b9d83f7e006dd6c809514ef803f44423fff4e799e3eca144e8c618L18-R26) [[4]](diffhunk://#diff-2713750720b9d83f7e006dd6c809514ef803f44423fff4e799e3eca144e8c618L33-R38) [[5]](diffhunk://#diff-2713750720b9d83f7e006dd6c809514ef803f44423fff4e799e3eca144e8c618L47)

### Module Imports:

* Adjusted the import statements in `Page.vue` to ensure all necessary modules are properly imported and organized. (`src/dispatch/static/dispatch/src/case/Page.vue`)
Screenshot 2024-10-08 at 5 04 42 PM
Screenshot 2024-10-08 at 5 04 29 PM

### CODE REVIEW

File: src/dispatch/static/dispatch/src/components/GenaiAnalysisDisplay.vue [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      <div v-for="(value, key) in analysis" :key="key" class="analysis-item">
+        <h3>
+          <b>{{ key }}</b>
+        </h3>
+        <div v-if="isObject(value)">
+          <!-- Render each sub-value with formatText to handle links and code formatting -->
+          <div
+            v-for="(subValue, subKey) in value"
+            :key="subKey"
+            v-html="formatText(subValue)"
```
Comment (wssheldon) [inline src/dispatch/static/dispatch/src/components/GenaiAnalysisDisplay.vue:14]: Do we need `v-html` here, I believe this is a risk of XXS vulnerabilities with it.

Comment (wssheldon) [reply src/dispatch/static/dispatch/src/components/GenaiAnalysisDisplay.vue:14]: Perhaps something like:

```vue
<template>
  <span>
    <template v-for="(part, index) in formattedParts" :key="index">
      <a v-if="part.type === 'link'" :href="part.href" target="_blank">{{ part.text }}</a>
      <code v-else-if="part.type === 'code'">{{ part.text }}</code>
      <span v-else>{{ part.text }}</span>
    </template>
  </span>
</template>

<script setup lang="ts">
import { computed } from 'vue';

const props = defineProps<{
  text: string;
}>();

interface TextPart {
  type: 'text' | 'link' | 'code';
  text: string;
  href?: string;
}

const formattedParts = computed<TextPart[]>(() => {
  const parts: TextPart[] = [];
  let remainingText = props.text;

  while (remainingText.length > 0) {
    const linkMatch = remainingText.match(/<([^|]+)\|([^>]+)>/);
    const codeMatch = remainingText.match(/`([^`]+)`/);

    if (!linkMatch && !codeMatch) {
      parts.push({ type: 'text', text: remainingText });
      break;
    }

    const linkIndex = linkMatch ? linkMatch.index! : Infinity;
    const codeIndex = codeMatch ? codeMatch.index! : Infinity;

    if (linkIndex < codeIndex) {
      if (linkIndex > 0) {
        parts.push({ type: 'text', text: remainingText.slice(0, linkIndex) });
      }
      parts.push({ type: 'link', href: linkMatch![1], text: linkMatch![2] });
      remainingText = remainingText.slice(linkIndex + linkMatch![0].length);
    } else {
      if (codeIndex > 0) {
        parts.push({ type: 'text', text: remainingText.slice(0, codeIndex) });
      }
      parts.push({ type: 'code', text: codeMatch![1] });
      remainingText = remainingText.slice(codeIndex + codeMatch![0].length);
    }
  }

  return parts;
});
</script>

<style scoped>
a {
  color: #1a73e8;
  text-decoration: underline;
}

code {
  font-family: monospace;
  background-color: #f5f5f5;
  padding: 0.2em 0.4em;
  border-radius: 4px;
}
</style>
```

Which could then be used like:

```vue
      <div v-for="(value, key) in analysis" :key="key" class="analysis-item">
        <h3>
          <b>{{ key }}</b>
        </h3>
        <div v-if="isObject(value)">
          <FormattedText
            v-for="(subValue, subKey) in value"
            :key="subKey"
            :text="subValue"
          />
        </div>
        <FormattedText v-else :text="value" />
      </div>
    </div>
```

Comment (mvilanova) [reply src/dispatch/static/dispatch/src/components/GenaiAnalysisDisplay.vue:14]: Yup, you're right that its usage could expose us to XSS attacks. Let me find a way to sanitize the contents.

**Comment by wssheldon** [comment]
I can make the change in a follow up PR. For the design, I'd suggest me change the `Timeline` table to `Overview` and move the GenAI summary into a Card that is displayed side by side with the timeline

**Review by wssheldon — APPROVED** [review]

**Pull request merged**